tools/code_format/check_format.py

```diff
@@ -103,6 +103,7 @@
 HISTOGRAM_SI_SUFFIX_REGEX = re.compile(r"(?<=HISTOGRAM\()[a-zA-Z0-9_]+_(b|kb|mb|ns|us|ms|s)(?=,)")
 TEST_NAME_STARTING_LOWER_CASE_REGEX = re.compile(r"TEST(_.\(.*,\s|\()[a-z].*\)\s\{")
 EXTENSIONS_CODEOWNERS_REGEX = re.compile(r'.*(extensions[^@]*\s+)(@.*)')
+DURATION_VALUE_REGEX = re.compile(r'duration\([0-9]+')
```

Comment:
Would this incorrectly flag other functions named `Something::duration(`?

Reply:
yeah, would be open to suggestions to make it more specific, maybe we could do `(SystemTime|MonotonicTime)::duration\([0-9]`, but if https://github.com/envoyproxy/envoy/blob/f9773269566002fb79593acc2d7089c897d67ca1/include/envoy/common/time.h#L15-L16 changes or there are additions made or if someone writes their own `std::chrono::system_clock::duration()` constructor that would break down

Reply:
Maybe write the regex as `r'([A-Za-z0-9_<>:]+)::duration\(` and if the captured value ends with `"Time"` or starts with `"std::chrono::"` report it. Should make it relatively easy to add exceptions if they become necessary. I recommend not checking for digits immediately following the open paren because it excludes variables and lines split by the formatter.

Reply:
We agree that we need to watch for a <token break>delimiter which was simple to do with \b. But the chances of a duration(123456) int value being line-folded is highly unlikely, enough so that it probably isn't worth catching the rare exception.

We do not care about using duration() in general, only in identifying cases where integers or floats are used without the context of explicit unit of measure. Variable names don't suffer this same issue, they can be as explicit or ambiguous as the coder chooses, and isn't something that envoy needs to or really can effectively oversee, based on our search of duration outside of the integer argument cases.

By definition, any "duration" method is going to be equally ambiguous in conjunction with an integer without the added context of a more precise method name. There are several trivial ways to add more context to the source code that will get any offending code past this rule, once we find a valid counterexample, this can be made more flexible later.

Reply:
Sounds good.